src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Listing/Listing.ts

```diff
@@ -294,6 +288,60 @@ export class Listing<Container extends ResourcesBase.Resource> {
     }
 }
 
+export var mapSwitchDirective = (
+    adhConfig : AdhConfig.IService
+) => {
+    return {
+        restrict: "E",
+        templateUrl: adhConfig.pkg_path + pkgLocation + "/MapSwitch.html"
+    };
+};
+
+export var filterAndSortDirective = () => {
+    return {
+        restrict: "E",
+        template: "<a data-ng-class=\"{ 'm-selected': data.showFilter }\" class=\"{{class}}\" href=\"\"\
+        data-ng-click=\"toggleFilter();\">{{ \"TR__FILTER\" | translate }}</a>\
+        <a data-ng-class=\"{ 'm-selected': data.showSort }\" class=\"{{class}}\" href=\"\"\
+        data-ng-click=\"toggleSort();\">{{ \"TR__SORT\" | translate }}</a>",
+        scope: {
+            class: "@",
+            data: "=?"
+        },
+        link: (scope) => {
+            scope.toggleFilter = () => {
+                scope.data.showSort = false;
+                scope.data.showFilter = !scope.showFilter;
+            };
+
+            scope.toggleSort = () => {
+                scope.data.showFilter = false;
+                scope.data.showSort = !scope.showSort;
+            };
+        }
+    };
+};
```

Comment:
The actual filter and sort options should be included in this directive (currently it is only the buttons that toggle showing them)